code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -590,14 +590,17 @@
 	if(.)
 		return
 
-	var/atom/movable/backup = get_spacemove_backup()
-	if(backup)
-		if(movement_dir && !backup.anchored)
-			if(backup.newtonian_move(turn(movement_dir, 180)))
+	var/atom/backup = get_spacemove_backup()
+	if(backup && movement_dir)
+		if(isturf(backup)) //get_spacemove_backup() already checks if a returned turf is solid, so we can just go
+			return TRUE
+		if(istype(backup, /atom/movable))
+			var/atom/movable/movable_backup = backup
+			if((!movable_backup.anchored) && (movable_backup.newtonian_move(turn(movement_dir, 180))))
 				step_silent = TRUE
 				if(return_drivers())
-					to_chat(occupants, "[icon2html(src, occupants)]<span class='info'>The [src] push off [backup] to propel yourself.</span>")
-		return TRUE
+					to_chat(occupants, "[icon2html(src, occupants)]<span class='info'>The [src] push off [movable_backup] to propel yourself.</span>")
+			return TRUE
```

Comment:
could you use early returns for the backup check?

Reply:
No because after this whole block, the proc does thruster checks. If I had an early `return FALSE`, thrusters would not work.